src/build-config.ts

```diff
@@ -128,9 +128,10 @@ async function buildInputConfig(
     entries,
     pkg,
     cwd,
-    tsOptions: { tsConfigPath, tsCompilerOptions },
+    tsOptions: { hasTsConfig, tsConfigPath, tsCompilerOptions },
```

Comment:
hasTsConfig should be identical but to `tsConfigPath` in boolean form, since when `tsConfigPath` is undefined it's without tsconfig